src/app/application.cpp

```diff
@@ -410,9 +422,12 @@ bool Application::isFileLoggerBackup() const
 
 void Application::setFileLoggerBackup(const bool value)
 {
-    if (m_fileLogger)
-        m_fileLogger->setBackup(value);
-    m_storeFileLoggerBackup = value;
+    if (isFileLoggerBackup() != value)
```

Comment:
It would be more efficient to add such guards in appropriate FileLogger setters instead. SettingValue lookups value by string key among all the settings each time you access it.

Reply:
> Or use `CacheSettingValue` instead?

It is mostly intended for those settings that are read multiple times during regular runtime (i.e. when they aren't changed).

Reply:
> Do you mean something like this?

No, of course!
1. Just keep corresponding setters of Application as they were before (i.e. without any guards against setting values)
2. Don't add any guards in those FileLogger setters where you just assign value to variable but only if changing value immediately causes some more or less expensive logic to be executed.